### PR TITLE
unify Usb and Wifi printing API

### PR DESCRIPTION
**Pull request by jobsorrow**

### CODE REVIEW

File: example/discover_printers/pubspec.yaml [hunk 1]
```diff
@@ -18,10 +18,10 @@ dependencies:
   intl: ^0.16.1
   qr_flutter: ^3.2.0
   path_provider: ^1.6.5
-  esc_pos_utils: ^1.0.0
-  # esc_pos_utils:
-  # path: ../../../esc_pos_utils
-  charset_converter: ^1.0.3
+  # esc_pos_utils: ^1.0.0
+  esc_pos_utils:
+    path: /home/test/esc_pos_utils
```
Comment (copsterr) [inline example/discover_printers/pubspec.yaml:23]: Is it able to use a relative path?

File: lib/src/printer.dart [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+class EscPosPrinter {
+  dynamic connectedDevice;
+
+  /// scan every UsbPrinter and NetworkPrinter then return the list of printers
+  Future<List> discover() async {
+    final String ip = await Wifi.ip;
+    final String subnet = ip.substring(0, ip.lastIndexOf('.'));
+    const int port = 9100;
```
Comment (copsterr) [inline lib/src/printer.dart:42]: why is it specifically port 9100?

File: lib/src/printer.dart [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    final wifiDeviceList = <dynamic>[];
+    final wifiDeviceStream =
+        await NetworkAnalyzer.discover2(subnet, port).toList();
+
+    for (final addr in wifiDeviceStream) {
+      if (addr.exists) {
+        print('Found device: ${addr.ip}');
```
Comment (copsterr) [inline lib/src/printer.dart:49]: remove print or comment it as debug

File: lib/src/printer.dart [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  void disconect({int delayMs}) async {
+    if (connectedDevice is UsbPrinter) {
+      await Escposprinter.closeConn;
+      connectedDevice = null;
+    } else if (connectedDevice is NetworkPrinter) {
+      connectedDevice.socket.destroy();
+      if (delayMs != null) {
+        await Future.delayed(Duration(milliseconds: delayMs), () => null);
+      }
+    }
```
Comment (copsterr) [inline lib/src/printer.dart:92]: If there're only 2 cases, you should use if then else or only if

Comment (jobsorrow) [reply lib/src/printer.dart:92]: if there's more make else scope and comment as todo

**Pull request merged**